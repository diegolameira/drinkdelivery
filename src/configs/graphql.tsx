import React from 'react';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';
import gql from 'graphql-tag';

interface Props {
  children: any;
}

export const ApolloWrapper = ({ children }: Props) => (
  <ApolloProvider client={client}>{children}</ApolloProvider>
);

export const client = new ApolloClient({
  uri:
    'https://803votn6w7.execute-api.us-west-2.amazonaws.com/dev/public/graphql'
});

export const allCategoriesSearch = gql`
  query allCategoriesSearch {
    allCategory {
      title
      id
    }
  }
`;

export const pocCategorySearch = gql`
  query pocCategorySearch($id: ID!, $search: String!, $categoryId: Int!) {
    poc(id: $id) {
      products(categoryId: $categoryId, search: $search) {
        productVariants {
          title
          description
          imageUrl
          price
        }
      }
    }
  }
`;

export interface ProductVariantsInterface {
  title: string;
  description: string;
  imageUrl: string;
  price: number;
}

export interface ProductsInterface {
  productVariants: ProductVariantsInterface[];
}

export interface PocInterface {
  products: ProductsInterface[];
}

export interface PocCategorySearchInterface {
  poc: PocInterface;
}

export const pocSearchMethod = gql`
  query pocSearchMethod(
    $now: DateTime!
    $algorithm: String!
    $lat: String!
    $long: String!
  ) {
    pocSearch(now: $now, algorithm: $algorithm, lat: $lat, long: $long) {
      __typename
      id
      status
      tradingName
      officialName
      deliveryTypes {
        __typename
        pocDeliveryTypeId
        deliveryTypeId
        price
        title
        subtitle
        active
      }
      paymentMethods {
        __typename
        pocPaymentMethodId
        paymentMethodId
        active
        title
        subtitle
      }
      pocWorkDay {
        __typename
        weekDay
        active
        workingInterval {
          __typename
          openingTime
          closingTime
        }
      }
      address {
        __typename
        address1
        address2
        number
        city
        province
        zip
        coordinates
      }
      phone {
        __typename
        phoneNumber
      }
    }
  }
`;
