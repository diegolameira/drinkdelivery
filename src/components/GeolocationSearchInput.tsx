import React, { Component } from 'react';
import { debounce } from 'lodash-es';

import Input from '@/components/Input';
import { styl } from '@/configs/theme';
import styled from 'styled-components/native';
import { ListView, TouchableHighlight } from 'react-native';

interface Props {
  label?: string;
  placeholder?: string;
}

interface State {
  current: any;
  input: string;
  suggestions: any[];
}

const SearchURL = (term: string) =>
  `https://maps.googleapis.com/maps/api/place/autocomplete/json?types=geocode&key=AIzaSyBH-JogSQqy0OPUnSf-b1selHP7VGCL7S8&input=${term}`;

const formattedSuggestion = structured_formatting => ({
  mainText: structured_formatting.main_text,
  secondaryText: structured_formatting.secondary_text
});

export default class GeolocationSearchInput extends Component<Props, State> {
  debouncedSearch: (state: State) => void;

  constructor(props: Props) {
    super(props);
    this.debouncedSearch = debounce(this.search, 200);
  }

  onChangeText = input => {
    this.setState(
      {
        input
      },
      () => this.debouncedSearch(this.state)
    );
  };

  search(state: State) {
    fetch(SearchURL(state.input))
      .then(response => response.json())
      .then(({ predictions }) => {
        this.setState({
          suggestions: predictions.map((p, idx) => ({
            id: p.id,
            description: p.description,
            placeId: p.place_id,
            active: idx === 0 ? true : false,
            index: idx,
            formattedSuggestion: formattedSuggestion(p.structured_formatting),
            matchedSubstrings: p.matched_substrings,
            terms: p.terms,
            types: p.types
          }))
        });
      });
  }

  onSuggestionSelect(suggestion: Suggestion) {
    this.setState({
      current: suggestion,
      input: suggestion.formattedSuggestion.mainText,
      suggestions: []
    });
  }

  componentDidMount() {}

  render = () => {
    const { suggestions, current } = this.state || {};
    return (
      <Wrapper focused={!!suggestions && suggestions.length}>
        <Input
          {...this.props}
          value={current ? current.formattedSuggestion.mainText : false}
          onChangeText={text => this.onChangeText(text)}
        />
        {suggestions ? (
          <SuggestionList
            items={suggestions}
            onSuggestionSelect={suggestion =>
              this.onSuggestionSelect(suggestion)
            }
          />
        ) : null}
      </Wrapper>
    );
  };
}

interface Suggestion {
  id: string;
  description: string;
  placeId: string;
  active: boolean;
  index: string;
  formattedSuggestion: any;
  matchedSubstrings: string;
  terms: any;
  types: any;
}

interface SuggestionListPropsInterface {
  items: Suggestion[];
  onSuggestionSelect: (suggestion: Suggestion) => void;
}

const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });

class SuggestionList extends React.Component<SuggestionListPropsInterface, {}> {
  __renderItem = (suggestion: Suggestion) => (
    <TouchableHighlight
      onPress={() => this.props.onSuggestionSelect(suggestion)}
    >
      <Text>{suggestion.formattedSuggestion.mainText}</Text>
    </TouchableHighlight>
  );
  render() {
    return (
      <SuggestionWrapper
        dataSource={ds.cloneWithRows(this.props.items)}
        renderRow={this.__renderItem}
      />
    );
  }
}

const Wrapper = styl(styled.View)`
  position: ${props => (props.focused ? 'absolute' : 'relative')};
  padding: ${props => (props.focused ? '30px' : '0')};
  top: 0;
  right: 0;
  left: 0;
  z-index: 9999;
  background: #fff;
`;

const SuggestionWrapper = styl(styled.ListView)`
`;

const Text = styl(styled.Text)`
  font-size: 20px;
  padding: 10px;
`;
