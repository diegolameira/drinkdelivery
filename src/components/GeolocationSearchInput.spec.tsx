/* eslint-env jest */

import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Theme } from '@/configs/theme';

import GeolocationSearchInput from './GeolocationSearchInput';

Enzyme.configure({ adapter: new Adapter() });

describe(`GeolocationSearchInput`, () => {
  const theme = new Theme();
  const component = renderer.create(<GeolocationSearchInput theme={theme} />);
  const wrapper = shallow(<GeolocationSearchInput theme={theme} />);
  let tree: ReactTestRendererJSON = component.toJSON();

  beforeEach(() => {});
  it('should render as expected', () => {
    const props = {
      label: 'Label',
      placeholder: 'placeholder'
    };
    expect(tree).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();

    tree.props = props;
    wrapper.setProps(props);

    expect(tree).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
  });
});
