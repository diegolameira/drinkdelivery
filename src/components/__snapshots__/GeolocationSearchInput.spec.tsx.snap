// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeolocationSearchInput should render as expected 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 15,
          "fontWeight": "300",
          "height": 0,
          "lineHeight": 15,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "opacity": 0.5,
        },
        undefined,
      ]
    }
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "borderColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#000",
          "fontSize": 20,
          "paddingBottom": 18,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 18,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`GeolocationSearchInput should render as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GeolocationSearchInput
    theme={
      Theme {
        "background": "#fff",
        "danger": "#DB0000",
        "foreground": "#000",
        "primary": "#FCD635",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onChangeText": [Function],
      "theme": Theme {
        "background": "#fff",
        "danger": "#DB0000",
        "foreground": "#000",
        "primary": "#FCD635",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onChangeText": [Function],
        "theme": Theme {
          "background": "#fff",
          "danger": "#DB0000",
          "foreground": "#000",
          "primary": "#FCD635",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GeolocationSearchInput should render as expected 3`] = `
<View
  label="Label"
  placeholder="placeholder"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 15,
          "fontWeight": "300",
          "height": 0,
          "lineHeight": 15,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "opacity": 0.5,
        },
        undefined,
      ]
    }
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "borderColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "color": "#000",
          "fontSize": 20,
          "paddingBottom": 18,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 18,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`GeolocationSearchInput should render as expected 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GeolocationSearchInput
    label="Label"
    placeholder="placeholder"
    theme={
      Theme {
        "background": "#fff",
        "danger": "#DB0000",
        "foreground": "#000",
        "primary": "#FCD635",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "label": "Label",
      "onChangeText": [Function],
      "placeholder": "placeholder",
      "theme": Theme {
        "background": "#fff",
        "danger": "#DB0000",
        "foreground": "#000",
        "primary": "#FCD635",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "label": "Label",
        "onChangeText": [Function],
        "placeholder": "placeholder",
        "theme": Theme {
          "background": "#fff",
          "danger": "#DB0000",
          "foreground": "#000",
          "primary": "#FCD635",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
